docs/how-to/partitions.rst

```diff
@@ -0,0 +1,136 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+In the below examples, we work with three partitions: ``default``, ``kernel``, and ``bar-baz``.  The ``default`` and ``kernel`` partitions do not have a namespace.  The ``bar-baz`` partition is part of the ``component`` namespace.
```

Comment:
Perhaps it would be easier to understand if you establish the concept of partitions first, mentioning that they're statically created by the application, and covering namespaced partitions after that (maybe in a separate  section) as a mechanism to do dynamic partitioning.

Reply:
Good suggestion!